src/ledger/LedgerTxnTrustLineSQL.cpp

```diff
@@ -30,23 +29,12 @@ getTrustLineStrings(AccountID const& accountID, Asset const& asset,
     {
         throw NonSociRelatedException("XLM TrustLine?");
     }
-    else if (accountID == getIssuer(asset))
+    else if (key.trustLine().accountID == getIssuer(asset))
     {
         throw NonSociRelatedException("TrustLine accountID is issuer");
     }
-
-    accountIDStr = KeyUtils::toStrKey(accountID);
-    if (asset.type() == ASSET_TYPE_CREDIT_ALPHANUM4)
-    {
-        assetCodeToStr(asset.alphaNum4().assetCode, assetCodeStr);
-        issuerStr = KeyUtils::toStrKey(asset.alphaNum4().issuer);
-    }
-    else if (asset.type() == ASSET_TYPE_CREDIT_ALPHANUM12)
-    {
-        assetCodeToStr(asset.alphaNum12().assetCode, assetCodeStr);
-        issuerStr = KeyUtils::toStrKey(asset.alphaNum12().issuer);
-    }
-    else
+    else if (asset.type() != ASSET_TYPE_CREDIT_ALPHANUM4 &&
```

Comment:
Practically speaking, this condition will never be satisfied. The current implementation of `getIssuer` will throw if `asset` is not `ALPHANUM4` or `ALPHANUM12` (but I think one of your other PRs changes the `getIssuer` check on line 32).

But my main question, should we even be checking this anymore? Anything other than native is a valid trust line, and the new schema will accommodate any asset.

Reply:
I actually removed that check in https://github.com/stellar/stellar-core/pull/3098. Would you prefer I remove it here instead?

Reply:
No, that's fine. There's no need to rearrange just for the sake of rearranging, since both are going to get merged in the near future.